homeassistant/components/zha/sensor.py

```diff
@@ -675,3 +677,83 @@ def _rm_rs_action(self) -> str | None:
         ):
             return CURRENT_HVAC_IDLE
         return CURRENT_HVAC_OFF
+
+
+@MULTI_MATCH(channel_names=CHANNEL_BASIC)
+class RSSISensor(Sensor, id_suffix="rssi"):
+    """RSSI sensor for a device."""
+
+    _state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _device_class: SensorDeviceClass = SensorDeviceClass.SIGNAL_STRENGTH
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+    _attr_entity_registry_enabled_default = False
+
+    @classmethod
+    def create_entity(
+        cls,
+        unique_id: str,
+        zha_device: ZhaDeviceType,
+        channels: list[ChannelType],
+        **kwargs,
+    ) -> ZhaEntity | None:
+        """Entity Factory.
+
+        Return entity if it is a supported configuration, otherwise return None
+        """
+        key = CHANNEL_BASIC + "_rssi"
```

Comment:
```suggestion
        key = f"{CHANNEL_BASIC}_{cls.id_suffix}"
```
Then you don't have to define it for every class?

Reply:
good call, need to make _unique_id_suffix public. On it.